ironfish-cli/src/commands/service/sync.ts

```diff
@@ -67,13 +67,19 @@ export default class Sync extends IronfishCommand {
 
     this.log('Connecting to node...')
 
-    const client = await this.sdk.connectRpc()
-
     const api = new WebApi({ host: apiHost, token: apiToken })
 
-    let head = args.head as string | null
+    const head = args.head as string | null
+
+    void this.syncTransactionGossip(api, flags.maxUpload)
+    await this.syncBlocks(api, head, flags.maxUpload)
+  }
+
+  async syncBlocks(api: WebApi, head: string | null, maxUpload: number): Promise<void> {
+    const client = await this.sdk.connectRpc()
```

Comment:
Not sure here but can/should we use one connection for both of these calls?

Reply:
updated to reuse the client